Projects/Features/PlaylistFeature/Sources/Reactors/PlaylistDetailReactor.swift

```diff
@@ -0,0 +1,215 @@
+        var newState = state
+
+        switch mutation {
+        case let .updateOrder(data):
+            newState.dataSource = [PlayListDetailSectionModel(model: 0, items: data)]
+
+        case let .fetchData(metadata):
+            newState.backupDataSource = metadata.list
+            newState.dataSource = metadata.list
+            newState.header = metadata.header
+
+        case .beginEdit:
+            newState.isEditing = true
+        case .save:
+            newState.isEditing = false
+        case let .changeSelectedState((data, count)), let .changeAllState((data, count)):
+            newState.dataSource = [PlayListDetailSectionModel(model: 0, items: data)]
+            newState.selectedItemCount = count
+        case .undo:
+            newState.dataSource = state.backupDataSource
+            newState.isEditing = false
+        }
+
+        return newState
+    }
+}
+
+// MARK: - Mutate
+
+private extension PlaylistDetailReactor {
+    /// 서버에서 데이터 불러오기
+    func fetchData() -> Observable<Mutation> {
+        return fetchPlayListDetailUseCase.execute(id: key, type: type)
+            .catchAndReturn(
+                PlayListDetailEntity(
+                    key: key,
+                    title: "",
+                    songs: [],
+                    image: "",
+                    image_square_version: 0,
+                    image_round_version: 0,
+                    version: 0
+                )
+            )
+            .asObservable()
+            .map { [weak self] result in
+
+                guard let self else {
+                    return PlaylistMetaData(
+                        list: [],
+                        header: PlayListHeader(title: "", songCount: "", image: "", version: 0)
+                    )
+                }
+                return PlaylistMetaData(
+                    list: [PlayListDetailSectionModel(model: 0, items: result.songs)],
+                    header: PlayListHeader(
+                        title: result.title,
+                        songCount: "\(result.songs.count)곡",
+                        image: self.type == .wmRecommend ? result.key : result.image,
+                        version: self.type == .wmRecommend ? result.image_square_version : result.version
+                    )
+                )
+            }
+            .map(Mutation.fetchData)
+    }
+
+    /// 순서 변경
+    func updateOrder(src: Int, dest: Int) -> Observable<Mutation> {
+        var tmp = (currentState.dataSource.first ?? PlayListDetailSectionModel(model: 0, items: [])).items
+        let target = tmp[src]
+        tmp.remove(at: src)
+        tmp.insert(target, at: dest)
+        return .just(.updateOrder(tmp))
+    }
+
+    /// 저장(서버)
+    func saveData() -> Observable<Mutation> {
+        let dataSource = currentState.dataSource[0].items.map { $0.id }
+        let backupDataSource = currentState.backupDataSource[0].items.map { $0.id }
```

Comment:
```suggestion
        guard let dataSource = currentState.dataSource.first?.items.map { $0.id },
                    let backupDataSource = currentState.backupDataSource.first?.items.map { $0.id }
        else {
            return .empty()
        }
```
[0]으로 바로 접근하는데 정말 만에하나 혹시나 crash 터질 위험성 있으니 first로 가져오는거 어떤가요?

Reply:
앗 데티일을 신경쓰지 못했네요 죄송합니다._

Reply:
적용